CHANGES.md

```diff
@@ -1,5 +1,17 @@
 ### Unreleased
 
+#### Improve and simplify LP lexer (2021-12-02)
+
+- allow nested comments (fix #710)
+- generate only one kind of identifiers
+- replace everywhere `%S` by `\"%s\"`
+- move checking compatibility with Bindlib of identifiers from lexer to scope
+- move `is_keyword` from `lexer` to `pretty`
+- move `package.ml` from `common/` to `parsing/`
+- change `Config.map_dir` field type to `(Path.t * string) list`,
+  `Library.add_mapping` type to `Path.t * string -> unit`
+  and Compile.compile argument `lm` type to `Path.t * string`
```

Comment:
Why? What does this modification allow? What's the impact on the command line interface?

Reply:
Before, the Path.t argument was a string and had to be parsed to convert it into a path. Doing this is cleaner and allows to simplify various things. In particular, we can now use escaped identifiers in package files and with the command decision-tree. package is moved to parsing since it is about package file parsing.

Reply:
For instance, I can almost remove the file escape.ml (https://github.com/Deducteam/lambdapi/pull/759/commits/65d00e3c3ce0c6b16f20a122192c4e2ef1657685). I think I will do it in a next commit.